github-distributed-owners: init at 0.1.11

cameroncuttingedge opened the pull request:
A tool for auto generating GitHub compatible CODEOWNERS files from OWNERS files distributed through the file tree.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Aleksanaa commented on pkgs/by-name/gi/github-distributed-owners/package.nix:
```diff
@@ -0,0 +1,27 @@
+  cargoHash = "sha256-pt/GoXF/uSU78pZqG8PgFe+tlbwZH2qpGQD7jgC52NM=";
+
+  meta = with lib; {
+    description = "A tool for auto generating GitHub compatible CODEOWNERS files from OWNERS files distributed through the file tree";
```
```suggestion
    description = "Generare GitHub CODEOWNERS files from OWNERS files distributed through the file tree";
```

(too long)

Aleksanaa commented on pkgs/by-name/gi/github-distributed-owners/package.nix:
```diff
@@ -0,0 +1,27 @@
+
+  cargoHash = "sha256-pt/GoXF/uSU78pZqG8PgFe+tlbwZH2qpGQD7jgC52NM=";
+
+  meta = with lib; {
+    description = "A tool for auto generating GitHub compatible CODEOWNERS files from OWNERS files distributed through the file tree";
+    homepage = "https://github.com/andrewring/github-distributed-owners";
+    license = licenses.mit;
+    maintainers = with maintainers; [ cameroncuttingedge ];
```
```suggestion
  meta = {
    description = "A tool for auto generating GitHub compatible CODEOWNERS files from OWNERS files distributed through the file tree";
    homepage = "https://github.com/andrewring/github-distributed-owners";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ cameroncuttingedge ];
```

Aleksanaa commented on pkgs/by-name/gi/github-distributed-owners/package.nix:
```diff
@@ -0,0 +1,27 @@
+  fetchFromGitHub,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "github-distributed-owners";
+  version = "0.1.11";
+
+  src = fetchFromGitHub {
+    owner = "andrewring";
+    repo = "github-distributed-owners";
+    rev = "v${version}";
```
```suggestion
rustPlatform.buildRustPackage (finalAttrs: {
  pname = "github-distributed-owners";
  version = "0.1.11";

  src = fetchFromGitHub {
    owner = "andrewring";
    repo = "github-distributed-owners";
    tag = "v${finalAttrs.version}";
```

Aleksanaa commented on maintainers/maintainer-list.nix:
This should be in a separated commit `maintainers: add cameroncuttingedge`, and before the commit adding the package

cameroncuttingedge committed:
maintainers: add cameroncuttingedge

cameroncuttingedge committed:
github-distributed-owners: init at 0.1.11

Update pkgs/by-name/gi/github-distributed-owners/package.nix

Co-authored-by: awwpotato <awwpotato@voidq.com>

Update pkgs/by-name/gi/github-distributed-owners/package.nix

Co-authored-by: awwpotato <awwpotato@voidq.com>

awwpotato commented on pkgs/by-name/gi/github-distributed-owners/package.nix:
```diff
@@ -0,0 +1,27 @@
+
+  meta = {
+    description = "Generate GitHub CODEOWNERS files from OWNERS files distributed through the file tree";
+    homepage = "https://github.com/andrewring/github-distributed-owners";
```
```suggestion
    homepage = "https://github.com/andrewring/github-distributed-owners";
    changelog = "https://github.com/andrewring/github-distributed-owners/releases/tag/v${finalAttrs.version}";
```

awwpotato commented on pkgs/by-name/gi/github-distributed-owners/package.nix:
```diff
@@ -0,0 +1,27 @@
+  };
+
+  cargoHash = "sha256-pt/GoXF/uSU78pZqG8PgFe+tlbwZH2qpGQD7jgC52NM=";
+
```
can be tested with `nix run nixpkgs#nix-update -- github-distributed-owners`, will make a pr you can merge when a new release is published
```suggestion

  passthru.updateScript = nix-update-script { };

```

cameroncuttingedge replied:
Thanks! Ran it locally and appear to work.

```bash
 nix run nixpkgs#nix-update -- github-distributed-owners
$ nix eval --json --impure --expr '
let
  
  pkgs = import "/Users/cameronmora/workspace/nixpkgs";
  args =  builtins.functionArgs pkgs;
  inputs = (if args ? system then { system = builtins.currentSystem; } else {}) //
           (if args ? overlays then { overlays = [ ]; } else {});
  pkg = (pkgs inputs)."github-distributed-owners";
  sanitizePosition = x: x;

  positionFromMeta = pkg: let
    parts = builtins.match "(.*):([0-9]+)" pkg.meta.position;
  in {
    file = builtins.elemAt parts 0;
    line = builtins.fromJSON (builtins.elemAt parts 1);
  };

  raw_version_position = sanitizePosition (builtins.unsafeGetAttrPos "version" pkg);

  position = if pkg ? isRubyGem then
    raw_version_position
  else if pkg ? isPhpExtension then
    raw_version_position
  else if (builtins.unsafeGetAttrPos "src" pkg) != null then
    sanitizePosition (builtins.unsafeGetAttrPos "src" pkg)
  else
    sanitizePosition (positionFromMeta pkg);
in {
  name = pkg.name;
  old_version = pkg.version or (builtins.parseDrvName pkg.name).version;
  inherit raw_version_position;
  filename = position.file;
  line = position.line;
  urls = pkg.src.urls or null;
  url = pkg.src.url or null;
  rev = pkg.src.rev or null;
  tag = pkg.src.tag or null;
  hash = pkg.src.outputHash or null;
  go_modules = pkg.goModules.outputHash or null;
  go_modules_old = pkg.go-modules.outputHash or null;
  cargo_deps = pkg.cargoDeps.outputHash or null;
  cargo_vendor_deps = pkg.cargoDeps.vendorStaging.outputHash or null;
  raw_cargo_lock =
    if pkg ? cargoDeps.lockFile then
      let
        inherit (pkg.cargoDeps) lockFile;
        res = builtins.tryEval (sanitizePosition {
          file = toString lockFile;
        });
      in
      if res.success then res.value.file else false
    else
      null;
  composer_deps = pkg.composerVendor.outputHash or null;
  composer_deps_old = pkg.composerRepository.outputHash or null;
  npm_deps = pkg.npmDeps.outputHash or null;
  pnpm_deps = pkg.pnpmDeps.outputHash or null;
  yarn_deps = pkg.yarnOfflineCache.outputHash or null;
  yarn_deps_old = pkg.offlineCache.outputHash or null;
  maven_deps = pkg.fetchedMavenDeps.outputHash or null;
  has_nuget_deps = pkg ? nugetDeps;
  mix_deps = pkg.mixFodDeps.outputHash or null;
  tests = builtins.attrNames (pkg.passthru.tests or {});
  has_update_script = pkg.passthru.updateScript or null != null;
  src_homepage = pkg.src.meta.homepage or null;
  changelog = pkg.meta.changelog or null;
  maintainers = pkg.meta.maintainers or null;
}' --extra-experimental-features 'flakes nix-command'
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
fetch https://github.com/andrewring/github-distributed-owners/releases.atom
Not updating version, already 0.1.11
Package maintainers:
  - Cameron Byte (@cameroncuttingedge)
$ git -C /Users/cameronmora/workspace/nixpkgs diff -- /Users/cameronmora/workspace/nixpkgs/pkgs/by-name/gi/github-distributed-owners/package.nix
No changes detected, skipping remaining steps
```

awwpotato commented:
all the commits related to the package should be squashed into one

awwpotato commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 418605`

Logs: https://github.com/awwpotato/nixpkgs-review-gha/actions/runs/15802495739


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>github-distributed-owners</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>github-distributed-owners</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>github-distributed-owners</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>github-distributed-owners</li>
  </ul>
</details>

0xda157 reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.

Aleksanaa commented on maintainers/maintainer-list.nix:
```diff
@@ -4071,6 +4071,12 @@
+  cameroncuttingedge = {
+    email = "buckets-taxiway5l@icloud.com";
+    github = "cameroncuttingedge";
+    githubId = 10954866;
```
Didn't notice that 😅 we should have CI check maintainer entries

Aleksanaa commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 418605`

Logs: https://github.com/Aleksanaa/nixpkgs-review-gha/actions/runs/15949380872


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>github-distributed-owners</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>github-distributed-owners</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>github-distributed-owners</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>github-distributed-owners</li>
  </ul>
</details>

Defelo reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.
